crates/edr_eth/src/remote/eth.rs

```diff
@@ -335,7 +335,10 @@ pub struct Block<TX> {
     /// mix hash
     pub mix_hash: B256,
     /// hash of the generated proof-of-work. null when its pending block.
-    #[serde(with = "crate::serde::optional_u64")]
+    #[serde(
+        serialize_with = "crate::serde::optional_u64::serialize_padded",
```

Comment:
EthereumJS expects padded nonces the lack of which made a test fail.

Reply:
I ran into this issue while upgrading to revm v3.5.0

The underlying error is that the nonce should be a `B64` instead of a `u64`. I think that also voids the need for a separate serialization function

Reply:
Thanks, fixed in [8286e23](https://github.com/NomicFoundation/hardhat/pull/4707/commits/8286e2359766e6aea7a9c821eec6fea25110378c)